MovieApp/MovieApp/Presenters/HomePagePresenter.swift

```diff
@@ -2,11 +2,14 @@ import Foundation
 
 final class HomePagePresenter {
         
-    func getPopularMovies(completion: @escaping (Result<[Movie]?, RequestError>) -> Void) {
+    func getPopularMovies(completion: @escaping (Result<[MovieViewModel]?, RequestError>) -> Void) {
```

Comment:
`getPopularMovies` has an `@escaping` parameter, so you need to be careful not to create retain cycles.

Reply:
I would need to add _[weak self]_ to the function call in the view controller so if this movie fetching lasted forever, I could close and delete the view controller from memory, because it's reference in completion block wouldn't count.

```
presenter.getPopularMovies { [weak self] result in
            switch result {
            case .success(let movies):
                self?.movies = movies
                DispatchQueue.main.async {
                    self?.filmsCollectionView.reloadData()
                }
            case .failure(let error):
                print("Error loading data: \(error)")
            }
        }
```